public/Aryan Suri/Sidebar/sidebar.js

```diff
@@ -1,30 +1,159 @@
+            // Converts the contents of the sourceForm input box to the units specified in the targetForm this.unit menu and puts the result in the targetForm input box.In other words, this is the heart of the whole script...
+            var propIndex;
+            var sourceIndex;
+            var sourceFactor;
+            var targetIndex;
+            var targetFactor;
+            var result;
+            // Start by checking which this.property we are working in...
+            propIndex = document.property_form.the_menu.selectedIndex;
+            // Let's determine what this.unit are we converting FROM (i.e. source) and the this.factor needed to convert that this.unit to the base this.unit.
+            sourceIndex = sourceForm.unit_menu.selectedIndex;
+            sourceFactor = this.factor[propIndex][sourceIndex];
+            // Cool! Let's do the same thing for the target this.unit - the units we are converting TO:
+            targetIndex = targetForm.unit_menu.selectedIndex;
+            targetFactor = this.factor[propIndex][targetIndex];
+            // Simple, huh? let's do the math: a) convert the source TO the base this.unit: (The input has been checked by the CalculateUnit function).
+            result = sourceForm.unit_input.value;
+            // Handle Temperature increments!
+            if (this.property[propIndex] == "Temperature") {
+                result = parseFloat(result) + this.tempIncrement[sourceIndex];
+            }
+            result = result * sourceFactor;
+            // not done yet... now, b) use the targetFactor to convert FROM the base this.unit
+            // to the target this.unit...
+            result = result / targetFactor;
+            // Again, handle Temperature increments!
+            if (this.property[propIndex] == "Temperature") {
+                result = parseFloat(result) - this.tempIncrement[targetIndex];
+            }
+            // Ta-da! All that's left is to update the target input box:
+            targetForm.unit_input.value = result;
+        }
+    };
     function readability() {
         $('section.mt-content-container p').css("font-size", sessionStorage.getItem("font_size") + "rem");
         $('section.mt-content-container').css("margin-left", sessionStorage.getItem("page_width") + "px");
         $('section.mt-content-container').css("margin-right", sessionStorage.getItem("page_width") + "px");
         $('section.mt-content-container p').css("text-align", sessionStorage.getItem("text_align"));
-        $("#size").val(sessionStorage.getItem("font_size"));
-        $("#slider-page-width").val(sessionStorage.getItem("page_width"));
-        $("#toggler-text").attr("class", "toggler");
+        $("#size").val(parseFloat(sessionStorage.getItem("font_size")));
+        $("#slider-page-width").val(parseFloat(sessionStorage.getItem("page_width")));
+        //$("#toggler-text").val(parseFloat(sessionStorage.getItem("font_size")!));
```

Comment:
Looking at your source code, I see why you are delaying this portion. If desired, you may want to wrap the justification code in a function for easier reuse.